toolkit/scripts/containerized-build/create_container_build.sh

```diff
@@ -0,0 +1,208 @@
+    echo ">>>> ERROR: $1"
+    switch_to_normal_text
+    echo ""
+}
+
+help() {
+echo "
+Usage:
+sudo make containerized-rpmbuild [REPO_PATH=/path/to/CBL-Mariner] [MODE=test|build] [VERSION=1.0|2.0] [MOUNTS= /src/path:/dst/path] [ENABLE_REPO=y]
+
+Starts a docker container with the specified version of mariner.
+
+Optional arguments:
+    REPO_PATH:      path to the CBL-Mariner repo root directory. default: "current directory"
+    MODE            build or test. default:"build"
+                        In 'test' mode it will use a pre-built mariner chroot image.
+                        In 'build' mode it will use the latest published container.
+    VERISION        1.0 or 2.0. default: "2.0"
+    MOUNTS          mount a directory into the container. Should be of the form '/src/dir:/dest/dir'. For multiple mounts, please use ',' as delimiter
+                        e.g. MOUNTS=/src/dir1:/dest/dir1,/src/dir2:/dest/dir2
+    ENABLE_REPO:    Set to 'y' to use local RPMs to satisfy package dependencies. default: "n"
+
+To see help, run 'sudo make containerized-rpmbuild-help'
+"
+}
+
+build_chroot() {
+    pushd "${repo_path}/toolkit"
+    echo "Building worker chroot..."
+    make graph-cache REBUILD_TOOLS=y > /dev/null
+    popd
+}
+
+build_tools() {
+    pushd "${repo_path}/toolkit"
+    echo "Building tools..."
+    make go-depsearch REBUILD_TOOLS=y > /dev/null
+    make go-grapher REBUILD_TOOLS=y > /dev/null
+    make go-specreader REBUILD_TOOLS=y > /dev/null
+    popd
+}
+
+build_graph() {
+    pushd "${repo_path}/toolkit"
+    echo "Building dependency graph..."
+    make workplan > /dev/null
+    popd
+}
+
+# exit if not running as root
+if [ "$EUID" -ne 0 ]; then
+  echo -e "\033[31mThis requires running as root\033[0m "
+  exit
+fi
+
+script_dir=$(realpath $(dirname "${BASH_SOURCE[0]}"))
+tmp_dir=${script_dir%'/toolkit'*}/build/containerized-rpmbuild/tmp
+mkdir -p ${tmp_dir}
+topdir=/usr/src/mariner
+enable_local_repo=false
+
+while (( "$#")); do
+  case "$1" in
+    -m ) mode="$2"; shift 2 ;;
+    -v ) version="$2"; shift 2 ;;
+    -p ) repo_path="$(realpath $2)"; shift 2 ;;
+    -mo ) extra_mounts="$2"; shift 2 ;;
+    -r ) enable_local_repo=true; shift ;;
+    -h ) help; exit 1 ;;
+    ? ) echo -e "ERROR: INVALID OPTION.\n\n"; help; exit 1 ;;
+  esac
+done
+
+# Assign default values
+[[ -z "${repo_path}" ]] && repo_path=${script_dir} && repo_path=${repo_path%'/toolkit'*}
+[[ ! -d "${repo_path}" ]] && { print_error " Directory ${repo_path} does not exist"; exit 1; }
+[[ -z "${mode}" ]] && mode="build"
+[[ -z "${version}" ]] && version="2.0"
+
+cd "${script_dir}"  || { echo "ERROR: Could not change directory to ${script_dir}"; exit 1; }
+
+# ==================== Setup ====================
+
+# Get Mariner GitHub branch at $repo_path
+repo_branch=$(git -C ${repo_path} rev-parse --abbrev-ref HEAD)
+
+#Generate splash text based on mode
+if [[ "${mode}" == "build" ]]; then
+    figlet "Mariner Builder!" > ${tmp_dir}/splash.txt
```

Comment:
Sounds good.
Just fyi cwize1 had [pointed out](https://github.com/microsoft/CBL-Mariner/pull/5855#discussion_r1283684132). 
I don't know if the tool can run on Mariner, but this would definitely be one of the hinderances. So, I'll remove figlet for now.